api/exchange.md

```diff
@@ -0,0 +1,59 @@
+# Exchange
+
+Exchange is the main contract that serves the mechanics around the tokens.
+
+### Overview of Exchange Functionality:
+
+This contract enables the exchange of tokens in a flexible manner, with various exchange mechanics available such as **[Claim](/admin/simple-mechanics/claim/)**, **[Grade](/admin/simple-mechanics/grade/)**, **Purchase**, etc. This functionality allows users to exchange one or multiple assets. However, to prevent unauthorized transactions, a trusted source (Server) must sign the transaction before it can be executed on the blockchain. Once the signature and additional information from the Server response are obtained, they can be combined with the Asset information to execute the exchange on the blockchain.
+
+#### Workflow of Purchase mechanics.
+![](/img/exchange_diagram.jpeg)
+
+
+### Signing a Purchase Transaction:
+
+To sign any transaction, we use the standard [EIP-712](https://eips.ethereum.org/EIPS/eip-712). This combines all parameters passed to the contract function into one string and hashes them. When executing the transaction, the address will be recovered from signature and contract will check this address for the appropriate minting rights. If the address has the necessary rights, the transaction is processed.
+
+To sign a transaction via the Server endpoint, several arguments must be provided, including the **account address**, the **referral address** (if not applicable, pass '0x'), and the **templateId**.
+
+> Different mechanics mey require different input. For Excample **Craft** require only **craftId**.
```

Comment:
may

>>>>
all mechanics requires account and optional referrer plus specific id of mechanic